CONTRIBUTING.rst

```diff
@@ -0,0 +1,678 @@
+    specific language governing permissions and limitations
+    under the License.
+
+.. contents:: :local:
+
+Contributing
+============
+
+Contributions are welcome and are greatly appreciated! Every little bit helps,
+and credit will always be given.
+
+
+Types of Contributions
+======================
+
+Report Bugs
+-----------
+
+Report bugs through `Apache
+Jira <https://issues.apache.org/jira/browse/AIRFLOW>`__.
+
+Please report relevant information and preferably code that exhibits the
+problem.
+
+Fix Bugs
+--------
+
+Look through the JIRA issues for bugs. Anything is open to whoever wants to
+implement it.
+
+Implement Features
+------------------
+
+Look through the `Apache
+JIRA <https://issues.apache.org/jira/browse/AIRFLOW>`__. for features.
+Any unassigned "Improvement" issue is open to whoever wants to implement
+it.
+
+Any unassigned "Improvement" issue is open to whoever wants to implement it.
+
+We've created the operators, hooks, macros and executors we needed, but we've
+made sure that this part of Airflow is extensible. New operators, hooks, macros
+and executors are very welcomed!
+
+Improve Documentation
+---------------------
+
+Airflow could always use better documentation, whether as part of the official
+Airflow docs, in docstrings, ``docs/*.rst`` or even on the web as blog posts or
+articles.
+
+Submit Feedback
+---------------
+
+The best way to send feedback is to open an issue on `Apache
+JIRA <https://issues.apache.org/jira/browse/AIRFLOW>`__.
+
+If you are proposing a feature:
+
+-   Explain in detail how it would work.
+-   Keep the scope as narrow as possible to make it easier to implement.
+-   Remember that this is a volunteer-driven project, and that contributions are
+    welcome :)
+
+Documentation
+=============
+
+The latest API documentation is usually available
+`here <https://airflow.apache.org/>`__..
+
+To generate a local version:
+
+1.  Set up an Airflow development environment.
+
+2.  Install the ``doc`` extra.
+
+.. code-block:: bash
+
+    pip install -e '.[doc]'
+
+
+3.  Generate and serve the documentation as follows:
+
+.. code-block:: bash
+
+    cd docs
+    ./build.sh
+    ./start_doc_server.sh
+
+
+Pull Request Guidelines
+=======================
+
+Before you submit a pull request (PR) from your forked repo, check that it meets
+these guidelines:
+
+-   Include tests, either as doctests, unit tests, or both, to your pull
+    request.
+
+    The airflow repo uses `Travis CI <https://travis-ci.org/apache/airflow>`__. to
+    run the tests and `codecov <https://codecov.io/gh/apache/airflow>`__. to track
+    coverage. You can set up both for free on your fork (see
+    `Travis CI Testing Framework <#travis-ci-testing-framework>`__. section below).
+    It will help you make sure you do not break the build with your PR and
+    that you help increase coverage.
+
+-   `Rebase your fork <http://stackoverflow.com/a/7244456/1110993>`__., squash
+    commits, and resolve all conflicts.
```

Comment:
Elena: do you think it's ok to keep these instructions in stackoverflow? Shall we add them to this doc?

Yes. Rebasing is quite standard procedure. Maybe some other link (for example this) can be used instead:
  https://www.atlassian.com/git/tutorials/merging-vs-rebasing

Reply:
@potiuk let's keep stackoverflow then. it's less generic.